client/i18n/locales/english/translations.json

```diff
@@ -527,7 +528,11 @@
     "start-project-err": "Something went wrong trying to start the project. Please try again.",
     "complete-project-first": "You must complete the project first.",
     "local-code-save-error": "Oops, your code did not save, your browser's local storage may be full.",
-    "local-code-saved": "Saved! Your code was saved to your browser's local storage."
+    "local-code-saved": "Saved! Your code was saved to your browser's local storage.",
+    "code-saved": "Your code was saved to the database. It will be here when you return.",
+    "code-save-error": "An error occurred trying to save your code.",
+    "challenge-save-too-big": "Sorry, you cannot save your code. The maximum size is {{max-size}} bytes. Your code is {{user-size}} bytes. Please make your code smaller and try again.",
+    "challenge-submit-too-big": "Sorry, you cannot submit your code. The a maximum size is {{max-size}} bytes. Your code is {{user-size}} bytes. Please make your code smaller and try again."
```

Comment:
```suggestion
    "challenge-save-too-big": "Sorry, we could not save your code.",
    "challenge-submit-too-big": "Sorry, you cannot submit your code.",
    "challenge-code-too-big": "Your code is {{user-size}} bytes. We allow a maximum of {{max-size}} bytes. Please make your code smaller and try again or request assistance on the forum."
```

Reply:
Of course you will need to update the usage to get the DRYing bit working.

Reply:
So is the idea here then that we would have a single flash message be something like this: `{challenge-save-too-big} {challenge-code-too-big}`? If that's the case, I would vote against it. When these get translated, they may need to adjust the way the sentences are arranged, or combine them somehow, in a way that it wouldn't make sense.

I'm in favor of adding the "or request assistance on the forum" part. Do you think we should make it a link to the forum? Or do we not want links in flash message?

Reply:
> When these get translated, they may need to adjust the way the sentences are arranged, or combine them somehow, in a way that it wouldn't make sense.

Ah - You are correct. I never thought about it that way, thanks for bringing that up. We should just reword the messages to be friendlier along the lines of my recommendation.

> Do you think we should make it a link to the forum? Or do we not want links in flash message?

Yes - that would be ideal.

Reply:
I ended up just changing the flash message to the suggestion and not having a link to the forum - there's some complexity there.

If we really want the link... 
option 1: would be to just use a plain text url. e.g. `request assistance at https://forum.freecodecamp.org/c/html-css`

option 2: the flash message component doesn't allow child elements at the moment - so we couldn't do `<a>the forum</a>` without adjustments. The component would have to be adjusted to use react-i18n's `Trans`. That might make creating a flash message look something like this:
```
createFlashMessage({
  type: 'danger',
  message: FlashMessages.ChallengeSaveTooBig,
  variables: { 'max-size': MAX_BODY_SIZE, 'user-size': bodySizeInBytes },
  children: [{ 'a' }]
})
```

and in the flash message component:
```
<Trans {variables.foreach(...)} i18nKey={message}>
{children.foreach(React.createElement)}
</Trans>
```

and in the translations:
```
"request assistance on the <0>forum.</0>"
```

That's probably not exactly what it would look like, but kind of close. I didn't think about it too hard, maybe there's a better way. If we want it to link to the forum, I would go with option 1 for now.

Reply:
For what it's worth, we do have a bare link in another flash message:

```js
"really-weird": "Something really weird happened, if it happens again, please consider raising an issue on https://github.com/freeCodeCamp/freeCodeCamp/issues/new",
```

Reply:
I noticed. This flash message will probably not be shown often so I don't think it's worth it to try and implement option 2 at the moment. Maybe in the future if we use more links or other elements in flash messages. That being said, do we want that plain URL, or just no link?

Reply:
I'd be okay with a link, just to `https://forum.freecodecamp.org` probably.

Reply:
went with that